pkg/operator/connectivitycheckcontroller/connectivity_check_controller.go

```diff
@@ -160,41 +178,66 @@ func (c *connectivityCheckController) Sync(ctx context.Context, syncContext fact
 		return nil
 	}
 
-	checks, err := c.podNetworkConnectivityCheckFn(ctx, syncContext)
+	var existingChecks []*v1alpha1.PodNetworkConnectivityCheck
+	if c.checkLister != nil {
+		existingChecks, err = c.checkLister.List(labels.Everything())
+		if err != nil {
+			return err
+		}
+	}
+
+	newChecks, err := c.podNetworkConnectivityCheckFn(ctx, syncContext)
 	if err != nil {
 		return err
 	}
 
 	pnccClient := c.operatorcontrolplaneClient.ControlplaneV1alpha1().PodNetworkConnectivityChecks(c.namespace)
-	for _, check := range checks {
-		existing, err := pnccClient.Get(ctx, check.Name, metav1.GetOptions{})
+	newCheckNames := sets.NewString()
+	for _, newCheck := range newChecks {
+		newCheckNames.Insert(newCheck.Name)
+		existing, err := pnccClient.Get(ctx, newCheck.Name, metav1.GetOptions{})
 		if err == nil {
-			if equality.Semantic.DeepEqual(existing.Spec, check.Spec) {
+			if value, ok := existing.Labels[managedByLabelKey]; ok &&
+				value == managedByLabelValue && equality.Semantic.DeepEqual(existing.Spec, newCheck.Spec) {
 				// already exists, no changes, skip
 				continue
 			}
 			updated := existing.DeepCopy()
-			updated.Spec = *check.Spec.DeepCopy()
+			updated.Spec = *newCheck.Spec.DeepCopy()
+			updated = setWithManagedByLabel(updated)
```

Comment:
This looks like a perfect use-case for a server-side apply that blindly applies the label in question.  @tkashem at some point the org has to learn, is this the moment?  If not, perhaps this is a good spot for the first migration?  Doing a server-side apply would eliminate the need to do the initial GET and would do an upsert for you.

Reply:
the server-side-apply is orthogonal to what this PR is doing, may be a separate PR that uses server-side-apply for connectivity checker?